config/src/main/java/io/confluent/common/config/ConfigUtils.java

```diff
@@ -48,8 +48,8 @@ public static Properties translateDeprecated(Properties props, String[][] synony
     // Properties object.
     HashSet<String> synonymSet = new HashSet<>();
     for (String[] synonymGroup: synonymGroups) {
-      for (String synonym : synonymGroup) {
-        synonymSet.add(synonym);
+      for (int i = 1; i < synonymGroup.length; i++) {
+        synonymSet.add(synonymGroup[i]);
```

Comment:
Bypassing this here means that we won't eliminate cases where the different synonyms are set to different values.  I think it would be better to fix this bug this way:

>      if (deprecated.isEmpty()) {
>        // No deprecated key(s) found.
+       newProps.put(target, props.get(target));
>        continue;
>      }